tests/pytests/test_parser.py

```diff
@@ -0,0 +1,311 @@
+    exp_res = '\
+VECTOR {\n\
+  @t1:INTERSECT {\n\
+    @t1:hello\n\
+    @t1:world\n\
+  }\n\
+} => {K=10 nearest vectors to `$B` in @v, AS `__v_score`}\n'
+    env.assertEqual(exp_res, res)
+
+def test_filters(env):
+    conn = getConnectionByEnv(env)
+    env.assertOk(conn.execute_command('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT', 't2', 'TAG', 'n', 'NUMERIC', 'g', 'GEO', 'v', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', '2','DISTANCE_METRIC', 'L2'))
+
+    res = conn.execute_command('FT.EXPLAIN', 'idx', 'very simple | @t:hello @t2:{ free\ world } (@n:[1 2]|@n:[3 4]) (@g:[1.5 0.5 0.5 km] -@g:[2.5 1.5 0.5 km])')
+    exp_res = '\
+UNION {\n\
+  INTERSECT {\n\
+    UNION {\n\
+      very\n\
+      +veri(expanded)\n\
+      veri(expanded)\n\
+    }\n\
+    UNION {\n\
+      simple\n\
+      +simpl(expanded)\n\
+      simpl(expanded)\n\
+    }\n\
+  }\n\
+  INTERSECT {\n\
+    @t:UNION {\n\
+      @t:hello\n\
+      @t:+hello(expanded)\n\
+    }\n\
+    TAG:@t2 {\n\
+      free\\ world\n\
+    }\n\
+    UNION {\n\
+      NUMERIC {1.000000 <= @n <= 2.000000}\n\
+      NUMERIC {3.000000 <= @n <= 4.000000}\n\
+    }\n\
+    INTERSECT {\n\
+      GEO g:{1.500000,0.500000 --> 0.500000 km}\n\
+      NOT{\n\
+        GEO g:{2.500000,1.500000 --> 0.500000 km}\n\
+      }\n\
+    }\n\
+  }\n\
+}\n'
+    env.assertEqual(exp_res, res)
+
+def test_combinations(env):
+    conn = getConnectionByEnv(env)
+    env.assertOk(conn.execute_command('FT.CREATE', 'idx', 'SCHEMA', 't', 'TEXT', 't2', 'TAG', 'n', 'NUMERIC', 'g', 'GEO', 'v', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', '2','DISTANCE_METRIC', 'L2'))
+
+    res = conn.execute_command('FT.EXPLAIN', 'idx', 'hello | "world" again', 'PARAMS', 2, 'B', '#blob#')
+    exp_res = '\
+UNION {\n\
+  UNION {\n\
+    hello\n\
+    +hello(expanded)\n\
+  }\n\
+  INTERSECT {\n\
+    world\n\
+    UNION {\n\
+      again\n\
+      +again(expanded)\n\
+    }\n\
+  }\n\
+}\n'
+    env.assertEqual(exp_res, res)
+
+    res = conn.execute_command('FT.EXPLAIN', 'idx', 'hello | -"world" again', 'PARAMS', 2, 'B', '#blob#')
+    exp_res = '\
+UNION {\n\
+  UNION {\n\
+    hello\n\
+    +hello(expanded)\n\
+  }\n\
+  INTERSECT {\n\
+    NOT{\n\
+      world\n\
+    }\n\
+    UNION {\n\
+      again\n\
+      +again(expanded)\n\
+    }\n\
+  }\n\
+}\n'
+    env.assertEqual(exp_res, res)
+
+    res = conn.execute_command('FT.EXPLAIN', 'idx', 'hello ~-"world" ~again', 'PARAMS', 2, 'B', '#blob#')
+    exp_res = '\
+INTERSECT {\n\
+  UNION {\n\
+    hello\n\
+    +hello(expanded)\n\
+  }\n\
+  OPTIONAL{\n\
+    NOT{\n\
+      world\n\
+    }\n\
+  }\n\
+  OPTIONAL{\n\
+    again\n\
+  }\n\
+}\n'
+    env.assertEqual(exp_res, res)
+
+def nest_exp(modifier, term, is_and, i):
+    if i == 1:
+        return '(@' + modifier + ':' + term + str(i) + ')'
+    return '(' + term + str(i) + (' ' if is_and else '|') + nest_exp(modifier, term, is_and, i - 1) + ')'
+
+def testDeepNesting(env):
```

Comment:
A bit nitpicking, but name is misleading - it is too deep
```suggestion
def testUnsupportedNesting(env):
```